$NetBSD$
Modified by the info from:
| Date: Sun, 22 Aug 2010 08:32:38 +0900 (JST)
| To: elips at heimat.gr.jp
| From: Hideyuki SHIRAI	<shirai at meadowy dot org>

--- egg/anthy.el.orig	2010-06-24 09:54:56.000000000 +0900
+++ egg/anthy.el	2010-08-21 18:43:32.012755200 +0900
@@ -168,6 +168,16 @@
 
 (defvar anthy-egg-agent-buffer-name " *anthy-egg*")
 
+(defmacro anthy-egg-static-if (cond then &rest else)
+  "Like `if', except that it evaluates COND at compile-time."
+  (if (eval cond) then `(progn  ,@else)))
+(put 'anthy-egg-static-if 'lisp-indent-function 2)
+
+(defun anthy-egg-process-kill-without-query (proc)
+  (anthy-egg-static-if (fboundp 'set-process-query-on-exit-flag)
+      (set-process-query-on-exit-flag proc nil)
+    (process-kill-without-query proc)))
+
 ;; Getting new context-descriptor, and returns environment with 'inuse' bit
 (defun anthy-egg-new-environment ()
   (if (null anthy-egg-proc)
@@ -182,7 +192,7 @@
 			 (append anthy-egg-agent-arguments
 				 (list "--utf8"))
 		       anthy-egg-agent-arguments)))
-	(set-process-query-on-exit-flag anthy-egg-proc nil)
+	(anthy-egg-process-kill-without-query anthy-egg-proc)
 	(set-process-coding-system anthy-egg-proc cs cs)
 	(set-process-sentinel anthy-egg-proc 'anthy-egg-proc-sentinel)
 	(set-marker-insertion-type (process-mark anthy-egg-proc) t)
@@ -637,7 +647,7 @@
 			    '("--append" "--utf8")
 			  '("--append"))))
       (when proc
-	(set-process-query-on-exit-flag proc nil)
+	(anthy-egg-process-kill-without-query proc)
 	(set-process-coding-system proc cs cs)
 	(set-process-sentinel proc
 			      (lambda (proc event)


